benchexec/tools/ultimate.py

```diff
@@ -204,113 +188,125 @@ def _requires_ultimate_data(self, executable):
         # all versions before 0.1.24 do not require ultimatedata
         return not (int(major) == 0 and int(minor) < 2 and int(patch) < 24)
 
-    def cmdline(self, executable, options, tasks, propertyfile=None, rlimits=None):
-        if rlimits is None:
-            rlimits = {}
-
-        self._uses_propertyfile = propertyfile is not None
-        if _OPTION_NO_WRAPPER in options:
-            # do not use old wrapper script even if property file is given
-            self._uses_propertyfile = False
-            propertyfile = None
-            options.remove(_OPTION_NO_WRAPPER)
+    def cmdline(self, executable, options, task, resource_limits):
+        data_model_param = get_data_model_from_task(
+            task,
+            {ILP32: ["--architecture", "32bit"], LP64: ["--architecture", "64bit"]},
+        )
+        combined_options = options
+        if data_model_param and not self._is_sublist(data_model_param, options):
```

Comment:
This check would mean if a user manually wants to override the architecture of a task by adding `--architecture something` to the command line, the tool-info module will still add `--architecture` as well, leading to duplicate parameters.

I would suggest to simplify this and simply check wether `--architecture` exists in `options`. This is also how the other tool-info modules are currently implemented.

Reply:
A warning like you added is of course even nicer for the user!